tests/profiles.spec.ts

```diff
@@ -1,23 +1,147 @@
+  await editProfile(page, profile);
+  await page.getByText("Security policies").click();
+
+  await setOption(page, "Prevent the instance from being deleted", "true");
+  await setOption(page, "Run the instance in privileged mode", "true");
+  await setOption(
+    page,
+    "Prevent instance file system from being UID/GID shifted",
+    "true"
+  );
+  await setInput(page, "Base host id", "Enter ID", "11");
+  await setInput(page, "Idmap size", "Enter number", "22");
+  await setOption(page, "Unique idmap usage", "true");
+  await setOption(page, "Allow /dev/lxd in the instance", "true");
+  await setOption(page, "Make /1.0/images API available", "true");
+  await setOption(page, "Enable secureboot", "true");
+  await saveProfile(page);
+
+  await assertReadMode(page, "Prevent the instance from being deleted Yes");
+  await assertReadMode(
+    page,
+    "Run the instance in privileged mode (Containers only) Allow"
+  );
+  await assertReadMode(
+    page,
+    "Prevent instance file system from being UID/GID shifted on startup (Containers only) Yes"
+  );
+  await assertReadMode(page, "Base host id (Containers only) 11");
+  await assertReadMode(page, "Idmap size (Containers only) 22");
+  await assertReadMode(page, "Unique idmap usage (Containers only) Yes");
+  await assertReadMode(
+    page,
+    "Allow /dev/lxd in the instance (Containers only) Yes"
+  );
+  await assertReadMode(
+    page,
+    "Make /1.0/images API available over /dev/lxd (Containers only) Yes"
+  );
+  await assertReadMode(page, "Enable secureboot (VMs only) true");
+
+  await deleteProfile(page, profile);
+});
+
+test("profile snapshots", async ({ page }) => {
+  const profile = randomProfileName();
+  await createProfile(page, profile);
+  await editProfile(page, profile);
+  await page.getByText("Snapshots").click();
+
+  await setInput(page, "Snapshot name", "Enter name pattern", "snap123");
+  await setInput(page, "Expire after", "Enter expiry expression", "3m");
+  await setOption(page, "Snapshot stopped instances", "true");
+  await setInput(page, "Schedule\t-\tLXD", "Enter cron expression", "@daily");
+  await saveProfile(page);
+
+  await assertReadMode(page, "Snapshot name pattern snap123");
+  await assertReadMode(page, "Expire after 3m");
+  await assertReadMode(page, "Snapshot stopped instances Yes");
+  await assertReadMode(page, "Schedule @daily");
+
+  await deleteProfile(page, profile);
+});
+
+test("profile cloud init", async ({ page }) => {
+  const profile = randomProfileName();
+  await createProfile(page, profile);
+  await editProfile(page, profile);
+  await page.getByText("Cloud init").click();
+
+  await setCodeInput(page, "Network config", "foo:\n" + " - abc");
+  await setCodeInput(page, "User data", "bar:\n" + " - def");
+  await setCodeInput(page, "Vendor data", "baz:\n" + " - ghi");
+  await saveProfile(page);
+
+  await assertCode(page, "foo:");
+  await assertCode(page, "bar:");
+  await assertCode(page, "baz:");
```

Comment:
This test is failing because it's trying to write that `foo` text in the network config field before it's loaded - see the test video, around 0:18:

[3ce50750618b5f9e510f0234fe8efc2bac6262dc.webm](https://user-images.githubusercontent.com/56583786/233929638-560bdb05-4b3f-489a-9cbe-69bde49bcda8.webm)

Reply:
Yes there was some logic to figure out when the editor is finished loading based on timeouts. I'll better wait for the class of the editor to appear.